.github/workflows/ubuntu.yml

```diff
@@ -5,59 +5,30 @@ on:
     paths-ignore:
     - '.github/workflows/**'
     - '!.github/workflows/ubuntu.yml'
+    - '!.github/install-clang-15.sh'
   pull_request:
     paths-ignore:
     - '.github/workflows/**'
     - '!.github/workflows/ubuntu.yml'
+    - '!.github/install-clang-15.sh'
 
 env:
   CFLAGS: -Wall -Werror
 
 jobs:
-  clang-15:
-    runs-on: ubuntu-22.04
+  test:
+    strategy:
+      matrix:
+        compiler: [clang-15, gcc]
+        os: [ubuntu-22.04, ubuntu-20.04]
+    runs-on: ${{ matrix.os }}
     env:
-      CC: /usr/bin/clang-15
-      CXX: /usr/bin/clang++-15
-      ASM: /usr/bin/clang-15
+      CC: ${{ matrix.compiler }}
+      DISTRO: ${{ matrix.os }}
```

Comment:
Can you, please, add a step below which will print compiler version? For "clang-15" it's eventually obvious, but for gcc it's not. And I do not think it's getting printed anywhere in autoconf configuraiton.

Reply:
Sure thing, added.